skabase/SKALogger/SKALogger/SKALogger.py

```diff
@@ -128,33 +130,86 @@ def Log(self, argin):
         # PROTECTED REGION END #    //  SKALogger.Log
 
     @command(
-    dtype_in='str', 
+    dtype_in='DevVarLongStringArray', 
     doc_in="Central logging level for selected devices", 
     )
     @DebugIt()
     def SetCentralLoggingLevel(self, argin):
         # PROTECTED REGION ID(SKALogger.SetCentralLoggingLevel) ENABLED START #
-        pass
+        CentralLoggingLevel = argin[0][:]
+        CentralLoggingDevice = argin[1][:]
+        i = 0
+        while i < len(CentralLoggingLevel[:]):
+            self.info_stream("%s,%s", CentralLoggingLevel[i], CentralLoggingDevice[i])
+            dev1 = DeviceProxy(CentralLoggingDevice[i])
+            dev1.centralLoggingLevel = CentralLoggingLevel[i]
+            property_names = ["logging_level",
+                              "logging_target",
+                              ]
+            dev_properties = dev1.get_property(property_names)
+            dev_properties["logging_level"] = ["DEBUG"]
+            dev_properties["logging_target"].append("device::central/logger/1")
+            dev1.put_property(dev_properties)
+            dev1.add_logging_target("device::central/logger/1")
+            i += 1
         # PROTECTED REGION END #    //  SKALogger.SetCentralLoggingLevel
 
     @command(
-    dtype_in='str', 
+    dtype_in='DevVarLongStringArray', 
     doc_in="Element logging level for selected devices", 
     )
     @DebugIt()
     def SetElementLoggingLevel(self, argin):
         # PROTECTED REGION ID(SKALogger.SetElementLoggingLevel) ENABLED START #
-        pass
+        ElementLoggingLevel = argin[0][:]
+        ElementLoggingDevice = argin[1][:]
+        i = 0
+        while i < len(ElementLoggingLevel[:]):
+            self.info_stream("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.debug("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.info("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.warning("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.error("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.fatal("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            dev1 = DeviceProxy(ElementLoggingDevice[i])
+            dev1.elementLoggingLevel = ElementLoggingLevel[i]
+            property_names = ["logging_level",
+                              "logging_target",
+                              "ElementLogger",
+                              "ElementLoglevel",
+                              "ServerHostName",
+                              ]
+            dev_properties = dev1.get_property(property_names)
+            dev_properties["logging_level"] = ["INFO"]
```

Comment:
As mentioned above, Remove hardcoded strings.

Reply:
Done.

Reply:
Have removed the properties setting lines from code. Yes, it keeps on adding the targets over and over again.